File: simpleFlowSolver2D/case.jl
Ima = 50; Jma = 50
delta = 1/Ima; nG = 1

Ifi = nG +1; Ifip=Ifi+1; Ifim=Ifi-1
Ila = Ima +nG; Ilap=Ila+1; Ilam=Ila-1
ImaAll = Ima + 2*nG
Jfi = nG +1; Jfip=Ifi+1; Jfim=Ifi-1
Jla = Jma +nG; Jlap=Jla+1; Jlam=Jla-1
JmaAll = Jma + 2*nG

DS = 1/4
CFL = 0.1
tMax = 0.1
nTMax = 10000
D = 1e-5
saveNT = 100

# velocity field: Vortex 
SF = 10.
x = LinRange( ((-ImaAll/2) +0.5) *delta, ((ImaAll/2) -0.5) *delta, ImaAll )
y = LinRange( ((-JmaAll/2) +0.5) *delta, ((JmaAll/2) -0.5) *delta, JmaAll )
Y = ones(length(x)) * y'
X = x * ones(length(y))'
R = sqrt.(X.^2 + Y.^2)
MV = SF./R
U = MV .* (Y ./R)
V = MV .* (-X ./R)

DeltaT = CFL *delta /(maximum(U))

# initialize scalar field
#Phi = Matrix{Float64}(undef, ImaAll, JmaAll)
Phi = zeros(ImaAll, JmaAll)
for i in Ifi:Ila
for j in Jfi:Jla
        Phi[i,j] = sin((i-nG-0.5)/Ima*2.0*pi) * sin((j-nG-0.5)/Jma*2.0*pi)
    end
end

Phie = zeros(size(Phi)...)
Phin = zeros(size(Phi)...)
Ue   = zeros(size(Phi)...)
Vn   = zeros(size(Phi)...)
FcX   = zeros(size(Phi)...)
FcY   = zeros(size(Phi)...)
FdX   = zeros(size(Phi)...)
FdY   = zeros(size(Phi)...)
PhiNew = zeros(size(Phi)...)

File: nfs2024_matlab/passiveScalarTransport2D/CASE.m
% # CASE.m ##############################################
% Programm: 		    	passiveScalarTranport2D
% Content:					define case, initiate fields
% Author:                   S.H. Tey, June 2024
% Email:                    sheu.tey@stud.uni-due.de
% Matriculation Number:     3117568
% -------------------------------------------------------
% initiate fields with the defined case
% =======================================================

% #######################################################
% ------- parameters: transported field initialization --
Phi = zeros(ImaAll, JmaAll);
for i = Ifi:Ila
    for j = Jfi:Jla
        Phi(i,j) = sin( (j-nG-0.5)/Ima *10*pi );
    end
end
% ------- parameters: velocity field initialization -----
Delta = 1/Ima;
x = linspace( ((-ImaAll/2) +0.5) *Delta, ...
        ((ImaAll/2) -0.5) *Delta, ImaAll )';
y = linspace( ((-JmaAll/2) +0.5) *Delta, ...
        ((JmaAll/2) -0.5) *Delta, JmaAll )';
X = ones(length(x), 1) * x';
Y = y * ones(length(y), 1)';
U = cos((X + 0.5)*2*pi) .* sin((Y + 0.5)*2*pi);
V = - sin((X + 0.5)*2*pi) .* cos((Y + 0.5)*2*pi);
% ------- parameters: other fields initialization -------
dims =  [ImaAll, JmaAll];
Phie =  zeros(dims);
Phin =  zeros(dims);
Ue =    zeros(dims);
Vn =    zeros(dims);
FcX =   zeros(dims);
FcY =   zeros(dims);
FdX =   zeros(dims);
FdY =   zeros(dims);
PhiNew =zeros(dims);
% #######################################################